java-checks/src/test/files/checks/security/CryptographicKeySizeCheck.java

```diff
@@ -0,0 +1,158 @@
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import javax.crypto.KeyGenerator;
+
+class A {
+
+  public KeyGenerator keyG;
+  public KeyPairGenerator keyPairG;
+  private KeyGenerator keyG2;
+
+  public A() throws NoSuchAlgorithmException {
+    keyG2 = KeyGenerator.getInstance("Blowfish");
+    keyPairG = KeyPairGenerator.getInstance("RSA");
+  }
+
+  public void foo1() throws NoSuchAlgorithmException {
+    KeyGenerator keyGen = KeyGenerator.getInstance("Blowfish");
+    keyGen.init(64); // Noncompliant [[sc=5;ec=20]] {{Use a key length of at least 128 bits.}}
+  }
+
+  public void foo2() throws NoSuchAlgorithmException {
+    KeyGenerator keyGen = KeyGenerator.getInstance("Blowfish");
+    int x = 64;
+    keyGen.init(x); // FN requires following variable's values with SE-based engine
+  }
+
+  public void foo3() throws NoSuchAlgorithmException {
+    String algorithm = "Blowfish";
+    KeyGenerator keyGen = KeyGenerator.getInstance(algorithm);
+    keyGen.init(64); // FN requires following variable's values with SE-based engine
+    keyGen.getAlgorithm();
+  }
+
+  public void foo4() throws NoSuchAlgorithmException {
```

Comment:
The names of the methods could be used to indicate what's unique to each test case: for example, this method could be named "key_assignment".